java-checks/src/test/files/checks/EqualsNotOverriddenInSubclassCheck.java

```diff
@@ -76,11 +76,16 @@ public boolean equals(Object obj) {
 
 class M {
   @Override
+  public boolean equals(Object obj) { return false; }
+}
+
+class N extends M {
+  @Override
   public final boolean equals(Object obj) {
     return false;
   }
 }
 
-class N extends M { // Compliant - M.equals() is final
+class O extends N { // Compliant - M.equals() is final
```

Comment:
Please update comment accordingly to your fix. Moreover, please add a dedicated testcase rather than modifying an existing one as this prevents to test for regression.

Reply:
Fixed it